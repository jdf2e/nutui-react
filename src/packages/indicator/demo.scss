.demo {
  .vertical_cell {
    height: 100px;
    width: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow: hidden;
  }
}
