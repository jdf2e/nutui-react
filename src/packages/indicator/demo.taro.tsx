import React from 'react'
import Taro from '@tarojs/taro'
import { useTranslate } from '@/sites/assets/locale/taro'
import { Cell, Indicator, ConfigProvider } from '@/packages/nutui.react.taro'
import Header from '@/sites/components/header'

interface T {
  ce5c5446: string
  c38a08ef: string
  c123sda1: string
  c38a08e4: string
}
const IndicatorDemo = () => {
  const [translated] = useTranslate<T>({
    'zh-CN': {
      ce5c5446: '基础用法',
      c38a08ef: '自定义节点',
      c38a08e4: '自定义',
      c123sda1: '竖向展示',
    },
    'zh-TW': {
      ce5c5446: '基礎用法',
      c38a08ef: '自定义节点',
      c38a08e4: '自定义',
      c123sda1: '豎向展示',
    },
    'en-US': {
      ce5c5446: 'Basic usage',
      c38a08ef: 'Custom Node',
      c38a08e4: 'Custom',
      c123sda1: 'Vertical display',
    },
  })

  const customTheme = {
    nutuiIndicatorColor: '#3768fa',
    nutuiIndicatorDotColor: '#ddd',
    nutuiIndicatorDotSize: '6px',
    nutuiIndicatorDotActiveSize: '24px',
  }
  return (
    <>
      <Header />
      <div className={`demo ${Taro.getEnv() === 'WEB' ? 'web' : ''}`}>
        <h2>{translated.ce5c5446}</h2>
        <Cell>
          <Indicator total={3} current={0} />
        </Cell>
        <Cell>
          <Indicator total={3} current={1} />
        </Cell>
        <Cell>
          <Indicator total={3} current={2} />
        </Cell>

        <h2>{translated.c38a08ef}</h2>
        <Cell>
          <Indicator total={6} current={5}>
            <div
              style={{
                display: 'inline-block',
                width: '14px',
                height: '14px',
                lineHeight: '14px',
                textAlign: 'center',
                fontSize: '12px',
                color: '#FFFFFF',
                border: '1px solid #FFFFFF',
                borderRadius: '50%',
                margin: '4px',
                background: `var(--nutui-color-primary)`,
                boxShadow: `0 0 1px 1px var(--nutui-color-primary)`,
              }}
            >
              {5}
            </div>
          </Indicator>
        </Cell>

        <h2>{translated.c38a08e4}</h2>
        <Cell>
          <ConfigProvider theme={customTheme}>
            <Indicator total={6} current={5} />
          </ConfigProvider>
        </Cell>

        <h2>{translated.c123sda1}</h2>
        <Cell>
          <Indicator total={6} current={5} direction="vertical">
            <div
              style={{
                display: 'inline-block',
                width: '14px',
                height: '14px',
                lineHeight: '14px',
                textAlign: 'center',
                fontSize: '12px',
                color: '#FFFFFF',
                border: '1px solid #FFFFFF',
                borderRadius: '50%',
                margin: '4px',
                background: `var(--nutui-color-primary)`,
                boxShadow: `0 0 1px 1px var(--nutui-color-primary)`,
              }}
            >
              {5}
            </div>
          </Indicator>
          <Indicator
            total={6}
            current={2}
            direction="vertical"
            style={{
              marginInlineStart: '50px',
            }}
          />
        </Cell>
      </div>
    </>
  )
}

export default IndicatorDemo
