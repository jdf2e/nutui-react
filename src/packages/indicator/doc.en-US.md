# Indicator

### Intro

Displays the progress of a task or process, often used for provisioning processes

### Install

```javascript
// react
import { Indicator } from '@nutui/nutui-react'

```
### Basic Usage
:::demo
```tsx
import  React from "react";
import { Indicator, Cell, Button, Row, Col } from '@nutui/nutui-react';

const App = () => {
  return (
    <div className="demo">
      <Cell>
        <Indicator size={3} current={3} />
      </Cell>
      <Cell>
        <Row>
          <Col span="12">
            <Button size="small" type="primary">
              Button
            </Button>
          </Col>
          <Col span="12">
            <Indicator block align="right" size={6} current={5} />
          </Col>
        </Row>
      </Cell>
    </div>
  );
};
export default App;
```
:::
### Block usage
When `block` is true, it will be displayed as a block-level element, and the alignment can be set through `align`
:::demo
```tsx
import  React from "react";
import { Indicator, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <div className="demo">
      <Cell>
        <Indicator block align="center" size={6} current={5} />
      </Cell>
      <Cell>
        <Indicator block align="left" size={6} current={1} />
      </Cell>
      <Cell>
        <Indicator block align="right" size={6} current={5} />
      </Cell>
    </div>
  );
};
export default App;
```
:::
### Do not add 0
:::demo
```tsx
import  React from "react";
import { Indicator, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <Cell>
      <Indicator fillZero={false} size={6} current={5} />
    </Cell>
  );
};
export default App;
```
:::

### Vertical display
:::demo
```tsx
import  React from "react";
import { Indicator, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <Cell>
      <view 
        style={{ height: '100px', width: '50%' }} 
      >
        <Indicator fillZero={false} size={6} current={5} vertical />
      </view>
      <view 
        style={{ height: '100px', width: '50%' }} 
      >
        <Indicator size={6} current={2} vertical />
      </view>
    </Cell>
  );
};
export default App;
```
:::
## API

### Props

| Attribute           | Description                             | Type                      | Default            |
|--------------|----------------------------------|--------|------------------|
| current  | current step               | Number | 1              |
| size       | step length                         | Number | 3               |
| block | Whether to enable block level layout     | Boolean | false |
| align | Alignment, only valid when block is true, optional values 'left', 'right', 'center' | String | left |
| fillZero     | Whether to add 0 in front of the singular number                      | Boolean | true        |
| vertical | Whether to display vertically     | Boolean | false |
