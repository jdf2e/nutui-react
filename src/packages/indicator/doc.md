# Indicator 指示器

## 介绍

显示一个任务或流程的进度，常用于开通流程。

## 安装

```tsx
import { Indicator } from '@nutui/nutui-react'
```

## 代码演示

### 基础用法

:::demo

```tsx
import  React from "react";
import { Indicator, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <div className="demo">
      <Cell>
        <Indicator total={3} current={2} />
      </Cell>
    </div>
  );
};
export default App;
```

:::

### 自定义节点

:::demo

```tsx
import  React from "react";
import { Indicator, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <Cell>
      <Indicator total={6} current={5}>
        <div className="number">{5}</div>
      </Indicator>
    </Cell>
  );
};
export default App;
```

:::

### 自定义颜色大小

:::demo

```tsx
import  React from "react";
import { Indicator, Cell, ConfigProvider } from '@nutui/nutui-react';

const customTheme = {
  nutuiIndicatorColor: '#3768fa',
  nutuiIndicatorDotColor: '#ddd',
  nutuiIndicatorDotSize: '8px',
  nutuiIndicatorDotActiveSize: '24px',
}

const App = () => {
  return (
    <Cell>
      <ConfigProvider theme={customTheme}>
        <Indicator total={6} current={5} />
      </ConfigProvider>
    </Cell>
  );
};
export default App;
```

:::

### 竖向展示

:::demo

```tsx
import  React from "react";
import { Indicator, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <Cell>
      <div className="vertical_cell">
        <Indicator total={6} current={5} direction="vertical">
          <div className="number">{5}</div>
        </Indicator>
      </div>
      <div className="vertical_cell">
        <Indicator total={6} current={2} direction="vertical" />
      </div>
    </Cell>
  );
};
export default App;
```

:::

## Indicator

### Props

| 属性 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| current | 当前步骤 | `number` | `0` |
| total | 步骤长度 | `number` | `3` |
| direction | 展示方向，默认为水平方向 | `horizontal` \| `vertical` | `horizontal` |

## 主题定制

### 样式变量

组件提供了下列 CSS 变量，可用于自定义样式，使用方法请参考 [ConfigProvider 组件](#/zh-CN/component/configprovider)。

| 名称 | 说明 | 默认值 |
| --- | --- | --- |
| \--nutui-indicator-color | 指示器焦点时色值 | `$primary-color` |
| \--nutui-indicator-dot-color | 指示器默认色值 | `$disable-color` |
| \--nutui-indicator-dot-size | 指示器尺寸 | `5px` |
| \--nutui-indicator-dot-active-size | 指示器焦点时尺寸 | `15px` |
| \--nutui-indicator-border-size | 指示器焦点时的border值 | `3px` |
| \--nutui-indicator-dot-margin | 指示器横向时的margin值 | `4px` |
| \--nutui-indicator-dot-vertical-margin | 指示器纵向时的margin值 | `4px` |