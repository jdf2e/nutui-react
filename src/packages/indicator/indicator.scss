.nut-indicator {
  display: flex;
  width: auto;
  flex-wrap: nowrap;
  align-items: center;

  &__dot {
    margin: 0 $indicator-dot-margin;

    &:first-child {
      margin-left: 0px;
    }

    &:last-child {
      margin-right: 0px;
    }
  }

  &__dot {
    display: inline-block;
    vertical-align: middle;
    width: $indicator-dot-size;
    height: $indicator-dot-size;
    border-radius: 50%;
    background-color: $indicator-dot-color;
  }

  &__active {
    width: $indicator-dot-active-size;
    border-radius: $indicator-border-radius-size;
    background: $indicator-color;
  }

  &__vertical {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    .nut-indicator__dot {
      margin: $indicator-dot-margin 0;

      &:first-child {
        margin-top: 0px;
      }

      &:last-child {
        margin-bottom: 0px;
      }
    }

    .nut-indicator__active {
      width: $indicator-dot-size;
      height: $indicator-dot-active-size;
    }
  }
}
