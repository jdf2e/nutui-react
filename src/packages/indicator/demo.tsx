import React from 'react'
import { Indicator } from './indicator'
import Cell from '@/packages/Cell'
import Row from '@/packages/row'
import Col from '@/packages/col'
import Button from '@/packages/button'

const IndicatorDemo = () => {
  return (
    <>
      <div className="demo">
        <h2>基础用法</h2>
        <Cell>
          <Indicator size={3} current={3}></Indicator>
        </Cell>
        <Cell>
          <Row>
            <Col span="12">
              <Button size="small" type="primary">
                主要按钮
              </Button>
            </Col>
            <Col span="12">
              <Indicator block={true} align="right" size={6} current={5}></Indicator>
            </Col>
          </Row>
        </Cell>
        <h2>block用法</h2>
        <Cell>
          <Indicator block={true} align="center" size={6} current={5}></Indicator>
        </Cell>
        <Cell>
          <Indicator block={true} align="left" size={6} current={1}></Indicator>
        </Cell>
        <Cell>
          <Indicator block={true} align="right" size={6} current={5}></Indicator>
        </Cell>
        <h2>不补0</h2>
        <Cell>
          <Indicator fillZero={false} size={6} current={5}></Indicator>
        </Cell>
      </div>
    </>
  )
}

export default IndicatorDemo
