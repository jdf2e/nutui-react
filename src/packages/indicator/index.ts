import { Indicator } from './indicator'

export type { IndicatorProps } from './indicator'
export default Indicator
