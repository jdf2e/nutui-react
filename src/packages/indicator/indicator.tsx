import React, { FunctionComponent, ReactNode, useEffect, useState } from 'react'
import './indicator.scss'
import bem from '@/utils/bem'
import classNames from 'classnames'

export interface IndicatorProps {
  size: number
  current: number
  block: boolean
  align: 'left' | 'right' | 'center'
  fillZero: boolean
}
const defaultProps = {
  size: 3,
  current: 1,
  block: false,
  align: 'center',
  fillZero: true,
} as IndicatorProps

export const Indicator: FunctionComponent<
  Partial<IndicatorProps> & React.HTMLAttributes<HTMLDivElement>
> = (props) => {
  const { size, current, block, align, fillZero, children, className, ...rest } = {
    ...defaultProps,
    ...props,
  }
  const b = bem('indicator')
  const classes = classNames(
    {
      [`${b('block')}`]: block,
      [`${b('align')}__${align}`]: block && align,
    },
    b('')
  )
  const renderEles = () => {
    let childs: ReactNode[] = []
    for (let item = 1; item <= size; item++) {
      childs.push(
        item == current ? (
          <div key={item} className={b('number')}>
            {fillZero && item < 10 ? `0${item}` : item}
          </div>
        ) : (
          <div key={item} className={b('dot')}></div>
        )
      )
    }
    return childs
  }
  useEffect(() => {}, [])
  return (
    <div className={`${classes} ${className}`} {...rest}>
      {renderEles()}
    </div>
  )
}

Indicator.defaultProps = defaultProps
Indicator.displayName = 'NutIndicator'
