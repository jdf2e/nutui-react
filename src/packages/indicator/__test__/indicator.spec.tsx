import * as React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import { Indicator } from '../indicator'
import Cell from '@/packages/cell'

test('should match snapshot', () => {
  const { asFragment } = render(<Indicator total={3} current={3} />)
  expect(asFragment()).toMatchSnapshot()
})

test('should be shown when passing size and current', () => {
  const { container } = render(
    <Cell>
      <Indicator total={3} current={3} />
    </Cell>
  )
  expect(container.querySelectorAll('.nut-indicator__dot')).toHaveLength(3)
  expect(container.querySelectorAll('.nut-indicator__active')).toHaveLength(1)
})

test('should be shown when custom node', () => {
  const { container } = render(
    <Cell>
      <Indicator total={6} current={5}>
        <div className="number">{5}</div>
      </Indicator>
    </Cell>
  )
  expect(container.querySelectorAll('.nut-indicator__dot')).toHaveLength(5)
})

test('should be shown when custom node', () => {
  const { container } = render(
    <Cell>
      <Indicator total={6} current={5}>
        <div className="number">{5}</div>
      </Indicator>
    </Cell>
  )
  expect(container.querySelectorAll('.nut-indicator__dot')).toHaveLength(5)
})
