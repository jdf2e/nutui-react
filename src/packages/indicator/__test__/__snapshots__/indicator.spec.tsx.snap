// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="nut-indicator"
  >
    <div
      class="nut-indicator-dot"
    />
    <div
      class="nut-indicator-dot"
    />
    <div
      class="nut-indicator-dot"
    />
  </div>
</DocumentFragment>
`;
